### DISCUSSION TITLE
POST request doesn't work with Hardhat task

### DISCUSSION THREAD

**Discussion by sellmon** [NomicFoundation/hardhat#3941] [Troubleshooting] [answered]
I'm using Hardhat on NextJS.
And I don't know how to run Hardhat task from my API

My API:
```
//pages/api/verify/index.ts
import type { NextApiRequest, NextApiResponse } from "next";

import { task } from "hardhat/config";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  try {
    const { contractAddress } = req.body;

    console.log("Args:", contractAddress);

    await task("verifyContract", "Verify factory contracts").setAction(
      async (taskArguments, hre, runSuper) => {
        console.log("Task verifyContract is running");
        const { contractAddress } = taskArguments;

        await hre.run("verify:verify", {
          address: contractAddress,
          constructorArguments: [
            "10000000000000000",
            "0x7600120a741b0969d41cf3804e373665cfd11b4",
          ],
        });

        console.log("Task verifyContract is done");
      }
    );

    console.log("Task skipped");

    res.status(200).json({
      message: "Server response: 200",
      contractAddress: contractAddress,
      taskCompleted: true,
    });
  } catch (error) {
    console.error("Error:", error);
    res
      .status(500)
      .json({ message: "Error" });
  }
}

```

My POST request:

```
const handleVerifyContract = async () => {
    const contractAddress = "0x4C51578AD78478c124B34Fb5812A87F81a47108c";

    try {
      console.log("Call function handleVerifyContract");

      const response = await fetch("/api/verify", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          contractAddress,
        }),
      });

      console.log("POST:", JSON.stringify({ contractAddress }));

      if (response.ok) {
        const data = await response.json();
        // setVerificationId(data.verificationId);
        console.log("Server response:", data.contractAddress);
      } else {
        throw new Error("Failed to verify contract.");
      }
    } catch (error) {
      console.error(error);
    }
  };
```

I send my post request and get this:
 - I send contractAddress correctly and get it from API response
 - On server side I get logs: "Args: 0x4C51578AD78478c124B34Fb5812A87F81a47108c" and Task skipped

I tried a lot of edits, and don't know why I can't run this task.
If I write something like "await run("verifyContract") or something like this - I'll get error:
```
You probably tried to import the "hardhat" module from your config or a file imported from it.
This is not possible, as Hardhat can't be initialized while its config is being defined.

To learn more about how to access the Hardhat Runtime Environment from different contexts go to https://hardhat.org/hre

```

**Comment by fvictorio** [comment, marked as answer]
You are defining a task inside a request handler. That... doesn't make sense?

I mean, `task` is used inside a Hardhat config to define tasks. Using it in any other context is almost surely a mistake.

If what you want is to verify a task programmatically, you have several options, but at least too come to mind:

- You can use [`child_process`](https://nodejs.org/api/child_process.html) to run `hardhat verify` with the right parameters.
- You could make Hardhat start the server and then use `hre.run` to run the verify task. This is probably a bad idea though, IMO in this context it's more natural to use a Hardhat task as a short-lived action.

I can't help you with the details of how to do this, but hopefully that sets you in the right track.

**Reply by sellmon** [reply]
Missed the message!

Thank you for the advice, I didn't know about child_process. It's a great solution! With it, I was able to quickly accomplish what I needed.
